packages/vue-query/src/useBaseQuery.ts

```diff
@@ -46,15 +45,14 @@ export function useBaseQuery<
   TQueryKey extends QueryKey,
 >(
   Observer: typeof QueryObserver,
-  arg1:
-    | TQueryKey
-    | UseQueryOptionsGeneric<TQueryFnData, TError, TData, TQueryKey>,
-  arg2:
-    | QueryFunction<TQueryFnData, UnwrapRef<TQueryKey>>
-    | UseQueryOptionsGeneric<TQueryFnData, TError, TData, TQueryKey> = {},
-  arg3: UseQueryOptionsGeneric<TQueryFnData, TError, TData, TQueryKey> = {},
+  genericOptions: UseQueryOptionsGeneric<
+    TQueryFnData,
+    TError,
+    TData,
+    TQueryKey
+  >,
 ): UseQueryReturnType<TData, TError> {
-  const options = computed(() => parseQueryArgs(arg1, arg2, arg3))
+  const options = computed(() => parseQueryArgs(genericOptions))
```

Comment:
should `parseQueryArgs` be renamed now, as it doesn't do any parsing anymore? Maybe `unrefQueryArgs` ? @DamianOsipiuk

Reply:
Done ✅

Reply:
I would use `cloneDeepUnref` directly here and in other places.
Also with some minor changes to types of `cloneDeepUnref` and `cloneDeep` we could get rid of many `as` assertions all over the codebase.
But i guess i could do that in another PR.
This should be good enough. 👍